internal/ansible/ansible_manager.go

```diff
@@ -0,0 +1,249 @@
+package ansible
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"git.sr.ht/~spc/go-log"
+	"github.com/apenella/go-ansible/pkg/execute"
+	"github.com/apenella/go-ansible/pkg/options"
+	"github.com/apenella/go-ansible/pkg/playbook"
+	ansibleResults "github.com/apenella/go-ansible/pkg/stdoutcallback/results"
+	"github.com/project-flotta/flotta-device-worker/internal/ansible/model/message"
+
+	"github.com/project-flotta/flotta-device-worker/internal/ansible/dispatcher"
+	pb "github.com/redhatinsights/yggdrasil/protocol"
+)
+
+// AnsibleManager handle ansible playbook execution
+type AnsibleManager struct {
+	dispatcherClient pb.DispatcherClient
+	// defined how to connect to hosts
+	ansiblePlaybookConnectionOptions *options.AnsibleConnectionOptions
+	// defined which should be the ansible-playbook execution behavior and where to find execution configuration.
+	ansiblePlaybookOptions *playbook.AnsiblePlaybookOptions
+	playbook               *playbook.AnsiblePlaybookCmd
+	ansibleDispatcher      *dispatcher.AnsibleDispatcher
+}
+
+type RequiredFields struct {
+	crcDispatcherCorrelationID, returnURL string
+	responseInterval                      uint64
+}
+
+const (
+
+	//Required field names
+	crcDispatcherAttribute    = "crc_dispatcher_correlation_id"
+	returnURLAttribute        = "return_url"
+	responseIntervalAttribute = "response_interval"
+	defaultResponseInterval   = 300
+
+	// Failure types
+	NotInstalled = "ANSIBLE_PLAYBOOK_NOT_INSTALLED"
```

Comment:
What do you think about changing this to `ANSIBLE_NOT_INSTALLED`

Reply:
It derives from [rhc-worker](https://github.com/RedHatInsights/rhc-worker-playbook/blob/8ddc5ccfc97290a021b4c4de673b92fedc38cbfb/rhc_worker_playbook/server.py#L86). 
But I agree, your suggestion makes the code clearer: I'll changed it.